src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -716,6 +716,11 @@ private unsafe SortKey IcuCreateSortKey(string source, CompareOptions options)
                     throw new PlatformNotSupportedException(GetPNSEWithReason("CreateSortKey", "non-invariant culture"));
                 return InvariantCreateSortKey(source, options);
             }
+#elif TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
```

Comment:
I think this `#if` should be moved after the `if ((options & ValidCompareMaskOffFlags) != 0)` exception guard or the `AssertExtensions.Throws<ArgumentException>("options", () => ci.GetSortKey(s1, CompareOptions.Ordinal));` test from `SortKeyMiscTest` should get temporarily disabled for apple mobile.